src/storage-operators/src/persist_source.rs

```diff
@@ -461,7 +461,27 @@ struct PendingWork {
     /// The time at which the work should happen.
     capability: Capability<(mz_repr::Timestamp, Subtime)>,
     /// Pending fetched part.
-    fetched_part: FetchedPart<SourceData, (), Timestamp, Diff>,
+    part: PendingPart,
+}
+
+enum PendingPart {
+    Blob(FetchedBlob<SourceData, (), Timestamp, Diff>),
+    Part(FetchedPart<SourceData, (), Timestamp, Diff>),
+}
```

Comment:
I assume that this exists so we can amortize the decoding across multiple steps, if we get a bunch of blobs at once. In which case, is it worth making this decoding explicitly consume fuel? I guess it's proportional to the work done now, more or less...

Reply:
I was more thinking that it's better from an allocation spike perspective to process them one at a time. Wasn't really sure how to map this into fuel, but it's at least under the yield-fn timer, which felt good enough to me.